File: Domain/Entities/ProjectOverview.cs
namespace Domain.Entities
{
    public class ProjectOverview
    {
        public Guid Id { get; set; }
        public string ProjectName { get; set; }
        public string ProjectLocation { get; set; }
        public string NatureProject { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Client { get; set; }
        public int Personnel { get; set; }
        public DateTime CreationDate { get; set; }
        public Guid CompanyId { get; set; }
        public CompanyUser Company { get; set; }
    }
}

File: client-app/src/app/master/models/projectOverview.ts
export interface ProjectOverview {
  id: string
  projectName: string
  projectLocation: string
  natureProject: string
  startDate: Date | null
  endDate: Date | null
  client: string
  personnel: number
  creationDate: Date | null
}

export class ProjectOverview implements ProjectOverview {
  constructor (init?: ProjectOverviewFormValues) {
    Object.assign(this, init)
  }
}

export class ProjectOverviewFormValues {
  id?: string = undefined
  projectName: string = ''
  projectLocation: string = ''
  natureProject: string = ''
  startDate: Date | null = null
  endDate: Date | null = null
  client: string = ''
  personnel: number = 0

  constructor (projectOverview?: ProjectOverviewFormValues) {
    if (projectOverview) {
      this.id = projectOverview.id
      this.projectName = projectOverview.projectName
      this.projectLocation = projectOverview.projectLocation
      this.natureProject = projectOverview.natureProject
      this.startDate = projectOverview.startDate
      this.endDate = projectOverview.endDate
      this.client = projectOverview.client
      this.personnel = projectOverview.personnel
    }
  }
}
